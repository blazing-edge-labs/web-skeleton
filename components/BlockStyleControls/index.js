import React from 'react'
import PropTypes from 'prop-types'
import { BLOCK_TYPES } from '../../constants/richTextEditor'
import StyleButton from '../StyleButton'

const BlockStyleControls = (props) => {
  const { editorState, onToggle } = props
  const selection = editorState.getSelection()
  const blockType = editorState
    .getCurrentContent()
    .getBlockForKey(selection.getStartKey())
    .getType()

  return (
    <div>
      {BLOCK_TYPES.map(type =>
        <StyleButton
          key={type.label}
          active={type.style === blockType}
          label={type.label}
          onToggle={onToggle}
          style={type.style}
        />
      )}
    </div>
  )
}

BlockStyleControls.propTypes = {
  editorState: PropTypes.object.isRequired,
  onToggle: PropTypes.func.isRequired,
}

export default BlockStyleControls
