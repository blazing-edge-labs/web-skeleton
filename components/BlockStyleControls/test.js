import React from 'react';
import renderer from 'react-test-renderer';
import BlockStyleControls from './';

describe('BlockStyleControls component snapshot', () => {
  it('renders basic required data', () => {
    const func = () => ({ getBlockForKey: () => ({ getType: () => {} }) });
    const tree = renderer.create(
      <BlockStyleControls
        editorState={{
          getCurrentContent: func,
          getSelection: () => ({ getStartKey: () => {} }),
        }}
        onToggle={() => {}}
      />
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
