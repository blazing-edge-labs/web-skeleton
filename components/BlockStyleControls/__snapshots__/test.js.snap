exports[`BlockStyleControls component snapshot renders basic required data 1`] = `
<div>
  <span
    onMouseDown={[Function]}>
    H1
  </span>
  <span
    onMouseDown={[Function]}>
    H2
  </span>
  <span
    onMouseDown={[Function]}>
    H3
  </span>
  <span
    onMouseDown={[Function]}>
    H4
  </span>
  <span
    onMouseDown={[Function]}>
    H5
  </span>
  <span
    onMouseDown={[Function]}>
    H6
  </span>
  <span
    onMouseDown={[Function]}>
    Blockquote
  </span>
  <span
    onMouseDown={[Function]}>
    UL
  </span>
  <span
    onMouseDown={[Function]}>
    OL
  </span>
  <span
    onMouseDown={[Function]}>
    Code Block
  </span>
</div>
`;
